shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -37,19 +38,91 @@ public static string GetTypeDisplayName(object item, bool fullName = true)
         public static string GetTypeDisplayName(Type type, bool fullName = true)
         {
             var sb = new StringBuilder();
-            ProcessTypeName(type, sb, fullName);
+            ProcessTypeName(type, sb, fullName ? NameFormatting.Full : NameFormatting.Short);
             return sb.ToString();
         }
 
-        private static void AppendGenericArguments(Type[] args, int startIndex, int numberOfArgsToAppend, StringBuilder sb, bool fullName)
+        public static string GetMethodDisplayName(MethodBase method, bool fullTypeName = true)
+        {
+            var sb = new StringBuilder();
+            ProcessMethodName(method, sb, fullTypeName);
+            return sb.ToString();
+        }
+
+        private static void ProcessMethodName(MethodBase method, StringBuilder sb, bool fullTypeName)
```

Comment:
The method printing can/should call into this class to print Types but anything else (like `void` or `.ctor`) should be kept separate from TypeNameHelper.

Reply:
Strictly speaking `typeof(void)` can also be an argument for `TypeNameHelper.GetDisplayName` method, and result like `void` more preferable than `System.Void`, although I understand that there are almost no usages of this type outside of method name output.

`.ctor` is just name of the method, and there is no special logic or constant for it,  although replacing it with the class name might be a good idea.

Reply:
Feel free to add `void` to the list regardless of changes for methodinfo is done or not.